client/js/nonprofits/donate/parseFields/customAmounts/index.ts

```diff
@@ -0,0 +1,17 @@
+// License: LGPL-3.0-or-later
+import JsonStringParser from './JsonStringParser';
+import { getDefaultAmounts } from '../../custom_amounts';
+import { CustomAmount } from '../customAmount';
+import parseNumberAmounts from './parseNumberAmounts';
+
+export default function parseCustomAmounts(amountsString: string): CustomAmount[] {
+  const defaultAmts = getDefaultAmounts().join();
+
+  if (amountsString.includes('{')) {
+    if (!amountsString.startsWith('[')) amountsString = `[${amountsString}`;
+    if (!amountsString.endsWith(']')) amountsString = `${amountsString}]`;
+    return new JsonStringParser(amountsString).results;
+  } else {
+    return parseNumberAmounts(amountsString || defaultAmts);
+  }
+}
```

Comment:
I think this.

The only tweak is I'd like an `index.spec.ts` too.

It's kind of redundant with `get-params.spec.js` but it gives us more certainty in the event get-params.js gets removed or replaced that the behavior of this file can be depended on.